nixpkgs_review/review.py

```diff
@@ -186,7 +188,10 @@ def start_review(self, attr: List[Attr], pr: Optional[int] = None) -> None:
         report = Report(attr)
         report.print_console(pr)
         report.write(self.builddir.path, pr)
-        nix_shell(report.built_packages(), self.builddir.path)
+        if self.batch:
+            sys.exit(0 if report.succeeded() else 1)
+        else:
+            nix_shell(report.built_packages(), self.builddir.path)
```

Comment:
To make this more general, I would instead propose to expose the `--run` argument of
`nix-shell`:

```
nix_shell(report.built_packages(), self.builddir.path, run=self.run)
```

This would also allow to interact with build artifacts more easily (they are stored in ./results.) before they are removed by nixpkgs-review again.
The exit status of `nix-shell` is also exposed to the user (when the command fail, python will also fail).

Reply:
I have not thought about on how to expose the build status to the command.
An additional environment variable within the nix-shell expression might help?

Reply:
I think that deviates slightly with the intention of the original PR. @bhipple just wants a command to tell him "Yes, this doesn't cause any regressions" or "No, theses now fail". 

I guess `--batch`,`--run args` could be a mutually exclusive switch. But --run still sounds like a different feature altogether.

I still think there's value in being able to have nix-review give you a "quick" yes/no. Also would be useful in CI applications

Reply:
Well in that case you would just do `--run true`.

Reply:
I would still say that it's less intuitive, similar to how you used to have to do `buildPhase = ":";` instead of `doBuild = false;` in nixpkgs

Reply:
Ok. Let's rename it to no-shell instead.